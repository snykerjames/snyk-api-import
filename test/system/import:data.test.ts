import { exec } from 'child_process';
import { sep, join } from 'path';
import { deleteFiles } from '../delete-files';
const main = './dist/index.js'.replace(/\//g, sep);

describe('`snyk-api-import import:data <...>`', () => {
  const OLD_ENV = process.env;
  afterAll(async () => {
    process.env = { ...OLD_ENV };
  });
  it('Shows help text as expected', (done) => {
    exec(
      `node ${main} import:data help`,
      {
        env: {
          PATH: process.env.PATH,
          GITHUB_TOKEN: process.env.GH_TOKEN,
          SNYK_LOG_PATH: __dirname,
        },
      },
      (err, stdout, stderr) => {
        if (err) {
          throw err;
        }
        expect(err).toBeNull();
        expect(stderr).toEqual('');
        expect(stdout.trim()).toMatchSnapshot();
        done();
      },
    );
  });

  it('Generates repo data as expected for Gitlab', (done) => {
    const orgDataFile = 'test/system/fixtures/org-data/orgs.json';
    exec(
      `node ${main} import:data --source=gitlab --integrationType=gitlab --sourceUrl=${process.env.TEST_GITLAB_BASE_URL} --orgsData=${orgDataFile}`,
      {
        env: {
          PATH: process.env.PATH,
          GITLAB_TOKEN: process.env.TEST_GITLAB_TOKEN,
          SNYK_LOG_PATH: __dirname,
        },
      },
      (err, stdout, stderr) => {
        if (err) {
          throw err;
        }
        expect(err).toBeNull();
        expect(stderr).toEqual('');
        expect(stdout.trim()).toMatch('Written the data to file');
        deleteFiles([join(__dirname, 'gitlab-import-targets.json')]);
        done();
      },
    );
  }, 20000);
  it('Generates repo data as expected for Bitbucket Server', (done) => {
    const orgDataFile =
      'test/system/fixtures/org-data/bitbucket-server-orgs.json';
    exec(
      `node ${main} import:data --source=bitbucket-server --integrationType=bitbucket-server --sourceUrl=${process.env.BBS_SOURCE_URL} --orgsData=${orgDataFile}`,
      {
        env: {
          PATH: process.env.PATH,
          BITBUCKET_SERVER_TOKEN: process.env.BBS_TOKEN,
          SNYK_LOG_PATH: __dirname,
        },
      },
      (err, stdout, stderr) => {
        if (err) {
          throw err;
        }
        expect(err).toBeNull();
        expect(stderr).toEqual('');
        expect(stdout.trim()).toMatch('Written the data to file');
        deleteFiles([join(__dirname, 'bitbucket-server-import-targets.json')]);
        done();
      },
    );
  }, 50000);
  it('Generates repo data as expected for Bitbucket Cloud', (done) => {
    const orgDataFile =
      'test/system/fixtures/org-data/bitbucket-cloud-orgs.json';
    exec(
      `node ${main} import:data --source=bitbucket-cloud --integrationType=bitbucket-cloud --orgsData=${orgDataFile}`,
      {
        env: {
          PATH: process.env.PATH,
          BITBUCKET_CLOUD_USERNAME: process.env.BBC_USERNAME,
          BITBUCKET_CLOUD_PASSWORD: process.env.BBC_PASSWORD,
          SNYK_LOG_PATH: __dirname,
        },
      },
      (err, stdout, stderr) => {
        if (err) {
          throw err;
        }
        expect(err).toBeNull();
        expect(stderr).toEqual('');
        expect(stdout.trim()).toMatch('Written the data to file');
        deleteFiles([join(__dirname, 'bitbucket-cloud-import-targets.json')]);
        done();
      },
    );
  }, 50000);
  it('Shows error when missing ', (done) => {
    exec(`node ${main} import:data --source=github`, (err, stdout, stderr) => {
      expect(err).toMatchSnapshot();
      expect(stderr).toMatchSnapshot();
      expect(stdout).toEqual('');
      done();
    });
  });
});
