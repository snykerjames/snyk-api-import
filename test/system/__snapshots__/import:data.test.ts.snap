// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`snyk-api-import import:data <...>\` Shows error when missing  1`] = `
[Error: Command failed: node ./dist/index.js import:data --source=github
index.js import:data

Generate data required for targets to be imported via API to create Snyk
projects.


Options:
  --version          Show version number                               [boolean]
  --help             Show help                                         [boolean]
  --orgsData         Path to organizations data file generated with
                     "orgs:create" command                            [required]
  --source           The source of the targets to be imported e.g. Github,
                     Github Enterprise, Gitlab, Azure. This will be used to make
                     an API call to list all available entities per org
    [required] [choices: "github", "github-enterprise", "gitlab", "azure-repos",
                      "bitbucket-server", "bitbucket-cloud"] [default: "github"]
  --sourceUrl        Custom base url for the source API that can list
                     organizations (e.g. Github Enterprise url)
  --integrationType  The configured integration type on the created Snyk Org to
                     use for generating import targets data. Applies to all
                     targets.
    [required] [choices: "github", "github-enterprise", "gitlab", "azure-repos",
                                          "bitbucket-server", "bitbucket-cloud"]

Missing required arguments: orgsData, integrationType
]
`;

exports[`\`snyk-api-import import:data <...>\` Shows error when missing  2`] = `
"index.js import:data

Generate data required for targets to be imported via API to create Snyk
projects.


Options:
  --version          Show version number                               [boolean]
  --help             Show help                                         [boolean]
  --orgsData         Path to organizations data file generated with
                     \\"orgs:create\\" command                            [required]
  --source           The source of the targets to be imported e.g. Github,
                     Github Enterprise, Gitlab, Azure. This will be used to make
                     an API call to list all available entities per org
    [required] [choices: \\"github\\", \\"github-enterprise\\", \\"gitlab\\", \\"azure-repos\\",
                      \\"bitbucket-server\\", \\"bitbucket-cloud\\"] [default: \\"github\\"]
  --sourceUrl        Custom base url for the source API that can list
                     organizations (e.g. Github Enterprise url)
  --integrationType  The configured integration type on the created Snyk Org to
                     use for generating import targets data. Applies to all
                     targets.
    [required] [choices: \\"github\\", \\"github-enterprise\\", \\"gitlab\\", \\"azure-repos\\",
                                          \\"bitbucket-server\\", \\"bitbucket-cloud\\"]

Missing required arguments: orgsData, integrationType
"
`;

exports[`\`snyk-api-import import:data <...>\` Shows help text as expected 1`] = `
"index.js import:data

Generate data required for targets to be imported via API to create Snyk
projects.


Options:
  --version          Show version number                               [boolean]
  --help             Show help                                         [boolean]
  --orgsData         Path to organizations data file generated with
                     \\"orgs:create\\" command                            [required]
  --source           The source of the targets to be imported e.g. Github,
                     Github Enterprise, Gitlab, Azure. This will be used to make
                     an API call to list all available entities per org
    [required] [choices: \\"github\\", \\"github-enterprise\\", \\"gitlab\\", \\"azure-repos\\",
                      \\"bitbucket-server\\", \\"bitbucket-cloud\\"] [default: \\"github\\"]
  --sourceUrl        Custom base url for the source API that can list
                     organizations (e.g. Github Enterprise url)
  --integrationType  The configured integration type on the created Snyk Org to
                     use for generating import targets data. Applies to all
                     targets.
    [required] [choices: \\"github\\", \\"github-enterprise\\", \\"gitlab\\", \\"azure-repos\\",
                                          \\"bitbucket-server\\", \\"bitbucket-cloud\\"]"
`;
